index.test.js

```diff
@@ -786,4 +786,118 @@ describe('q-helper', () => {
+                        notRequired: ['a', 'b', 'c']
+                    }
+                };
+                const arrayToDelete = ['b'];
+                const expected = {
+                    a: '1',
+                    required: ['a', 'c'],
+                    c: {
+                        a: '3',
+                        required: ['a', 'c'],
+                        notRequired: ['a', 'b', 'c']
+                    }
+                };
+                const actual = qHelper.removeSchemaElements(json, arrayToDelete);
+
+                expect(expected).toEqual(actual);
+            });
+        });
+
+        describe('isValidCompiledDocument', () => {
+            it('should return true if a schema is valid', () => {
+                const qTemplateHelper = createQuestionnaireTemplateHelper({
+                    qSchema: {
+                        type: 'object',
+                        properties: {
+                            foo: {
+                                type: 'integer'
+                            },
+                            bar: {
+                                type: 'string'
+                            },
+                            biz: {
+                                type: 'object',
+                                properties: {
+                                    a: {
+                                        type: 'string'
+                                    },
+                                    b: {
+                                        type: 'string'
+                                    },
+                                    c: {
+                                        type: 'string'
+                                    }
+                                },
+                                required: ['a', 'b', 'c']
+                            },
+                            sections: {
+                                type: 'object'
+                            },
+                            routes: {
+                                type: 'object',
+                                properties: {
+                                    states: {
+                                        type: 'object'
+                                    }
+                                }
+                            }
+                        },
+                        required: ['foo'],
+                        additionalProperties: false
+                    },
+                    questionnaireTemplate: {
+                        foo: 1,
+                        bar: 'abc',
+                        biz: {
+                            a: 'foo',
+                            c: 'bar'
+                        },
+                        sections: {},
+                        routes: {
+                            states: {}
+                        }
+                    }
+                });
+
+                expect(qTemplateHelper.isValidCompiledDocument(['b'])).toEqual(true);
```

Comment:
At a glance this test result is confusing to be. The array `['b']` is to check if there are any instances of an object property with the name `b`, yeah? But I can see properties called `b` within this JSON and the test is passing.